src/plugins/intel_gpu/src/graph/graph_optimizer/concat_input_order.cpp

```diff
@@ -211,8 +211,12 @@ void concat_input_order::run(program& p) {
         auto mutable_prim = std::const_pointer_cast<concatenation>(prim);
         mutable_prim->input = new_input_info;
         // Correct users for shuffled features
-        for (auto& user : concat_node.get_users()) {
-            shuffle_features(*user, shuffled_ranges, p.get_stream());
+        try {
+            for (auto& user : concat_node.get_users()) {
+                shuffle_features(*user, shuffled_ranges, p.get_stream());
+            }
+        } catch (std::invalid_argument& e) {
```

Comment:
[random spot] please add test

Reply:
Added the unit test, thanks for your comment.